src/Command/SetIssueTrackerTicketCommand.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace App\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use App\Entity\DatasetSubmission;
+use App\Entity\Dataset;
+
+/**
+ * This command sets the issue ticket on a datset (submisison).
+ */
+class SetIssueTrackerTicketCommand extends Command
+{
+    /**
+     * The Command name.
+     *
+     * @var string $defaultName
+     */
+    protected static $defaultName = 'pelagos:dataset-set-issue-ticket';
+
+    /**
+     * A Doctrine ORM EntityManager instance.
+     *
+     * @var EntityManagerInterface $entityManager
+     */
+    protected $entityManager;
+
+    /**
+     * Class constructor for dependency injection.
+     *
+     * @param EntityManagerInterface $entityManager A Doctrine EntityManager.
+     */
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+        parent::__construct();
+    }
+
+    /**
+     * Symfony command config section.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Sets the issue tracker ticket on a dataset.')
+            ->addOption('csvfile', null, InputOption::VALUE_REQUIRED, 'Filename of csv containing UDI, ticket')
+            ;
+    }
+
+    /**
+     * Symfony command execution section.
+     *
+     * @param InputInterface  $input  Command args.
+     * @param OutputInterface $output Output txt.
+     *
+     * @return integer Return code.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+        $filename = $input->getOption('csvfile');
+
+        if (!file_exists($filename)) {
+            $io->error('File not found: ' . $filename);
+            return 1;
+        }
+
+        if (($fileHandle = fopen($filename, "r")) !== false) {
+            while (($data = fgetcsv($fileHandle, 100, ",")) !== false) {
+                $udi = trim($data[0]);
+                $issueTrackingTicket = trim($data[1]);
+                $dataset = $this->entityManager->getRepository(Dataset::class)->findOneBy(array('udi' => $udi));
+                if (!($dataset instanceof Dataset)) {
+                    $io->warning("Could not find a dataset with UDI ($udi)");
+                    continue;
+                } else {
+                    if ($issueTrackingTicket === '') {
+                        $io->note("Removing Issue Tracking Ticket for Dataset $udi.");
+                        $dataset->setIssueTrackingTicket(null);
```

Comment:
That was my intent, to allow a previous value to be cleared by leaving it blank in the CSV. I'll ask the PO to be sure.

Reply:
I asked the PO and he said that if nothing is present in the ticket field, populate the value with nothing/null, overwriting the potential existing value with the blank.

Reply:
PO decided it was best to just throw an error for a CSV entry missing the ticket number instead of assuming the intention to null out an existing value. DRPMs will have to use the interface to null out an existing value instead. The tool will still replace an existing value with something else.